gazpacho/soup.py

```diff
@@ -98,6 +98,20 @@ def find(self, tag, attrs=None):
         self.group = 0
         self.groups = []
         super().feed(self.html)
-        if len(self.groups) == 1:
-            return self.groups[0]
         return self.groups
+
+    def find_one(self, tag, attrs=None):
+        '''Find a tag with optional attributes
+
+        - tag (str): HTML tag to find
+        - attrs (dict, optional): Attributes within tag to match
+        '''
+        self.tag = tag
+        self.attrs = attrs
+        self.count = 0
+        self.group = 0
+        self.groups = []
```

Comment:
Hey @OsamaNabih thanks for taking a stab at this! 
I [wrote](https://github.com/maxhumber/gazpacho/issues/3#issuecomment-538047416) in the issue just like a minute before you submitted the PR that I have a sketch of something like this already in 0.8 but that I'm not sure it's the right way to do it. I think we have to go into the HTMLParser and make some changes there...

Reply:
I'm not sure if there's another way to go about it. The functions have similar initialization but need to be kept logically separate for interfacing with the class. Not all replications can be avoided imo and this is kinda minor. Maybe wrap up the lines in a utility function for initialization but other than that I don't know.

Reply:
What would you change in HTMLParser such as to avoid this?

Reply:
I'm imagining some if-else control flow logic in handle_starttag!